### PR TITLE
Add instances for Scientific

### LINKED ISSUE #143: Field instances for Scientific
Would it be worth adding a dependency on scientific and having ToField/FromField instances for the Scientific type?

### PR DESCRIPTION
**Pull request by ivan-m**
Note that scientific was already a transitive dependency.

Closes #143

### CODE REVIEW

File: Data/Csv/Conversion.hs [hunk 1]
```diff
@@ -715,6 +716,18 @@ instance ToField Char where
     toField = toField . T.encodeUtf8 . T.singleton
     {-# INLINE toField #-}
 
+-- | Accepts the same syntax as 'rational'. Ignores whitespace.
+instance FromField Scientific where
+  parseField s = case parseOnly (ws *> A8.scientific <* ws) s of
+                   Left err -> typeError "Scientific" s (Just err)
+                   Right n  -> pure n
+  {-# INLINE parseField #-}
```
Comment (hvr) [inline Data/Csv/Conversion.hs]: note to myself: at some point I should figure out whether those INLINEs sprinkled throughout cassava are really a good idea, and whether we shouldn't rather use INLINEABLE...

File: cassava.cabal [hunk 2]
```diff
@@ -65,12 +65,13 @@ Library
...
   Build-depends:
     array >= 0.4 && < 0.6,
-    attoparsec >= 0.10.2 && < 0.14,
+    attoparsec >= 0.11.3.0 && < 0.14,
     base >= 4.5 && < 5,
     bytestring >= 0.9.2 && < 0.11,
     containers >= 0.4.2 && < 0.6,
     deepseq >= 1.1 && < 1.5,
     hashable < 1.3,
+    scientific >= 0.3.4.7,
```
Comment (hvr) [inline cassava.cabal]: upper bound please

Comment (ivan-m) [inline cassava.cabal]: I left it off originally as I only import it to get the actual data type itself, and I figured it was unlikely to move from `Data.Scientific.Scientific`.

File: Data/Csv/Conversion.hs (hunk 1 above)
Comment (ivan-m) [inline Data/Csv/Conversion.hs]: Yes, I've wondered the same thing with a lot of my code and the heavy INLINE hammers.

**Comment by ivan-m** [comment]
@hvr done

**Comment by hvr** [comment]
thanks!

**Pull request merged**